src/platform_linux/platform_inline.h

```diff
@@ -415,19 +442,63 @@ platform_aligned_malloc(const platform_heap_id UNUSED_PARAM(heap_id),
     * Note that since this is inlined, the compiler will turn the constant
     * (power of 2) alignment mod operations into bitwise &
     */
-   const size_t padding = (alignment - (size % alignment)) % alignment;
-   return aligned_alloc(alignment, size + padding);
+   const size_t padding  = platform_align_bytes_reqd(alignment, size);
```

Comment:
I think the way it's coded, it still achieves what we want; i.e., to ensure that the returned 'ptr' is always aligned.

This is the way I argued to myself that this scheme is acceptable.

Let's consider cache-line alignment is 64 bytes.

Say, we did not do this padding up business for every allocation request. Then, after the very 1st allocation of 60 bytes, the `next-addr` will be 60. For the next request, we would have to move the `next-addr` ahead to offset 64, so the returned memory ptr is correctly aligned.

The same thing is being achieved by rounding-up the previous memory request from 60 bytes to 64 bytes.

Essentially, every memory allocation ensures that the `next-addr` ptr is left correctly aligned for the next memory allocation request.

It wastes a tiny bit of space for the very last memory allocation, but for every allocation request before that, I think the net memory usage (wastage) is the same in both schemes.

Hope this makes sense.

Reply:
For the non-shmem allocations, this code is unnecessary -- the alignment parameter passed to the lower-level allocator will ensure the requested alignment.

For the shmem allocator, the approach in this code will not always work.  Suppose the `next-addr` starts out 64-byte aligned (which it presumably does).  Suppose the first two allocations performed are:
1. allocate some number of bytes with 32-byte alignment
2. allocate some number of bytes with 64-byte alignment.
The current code will will round up the first allocation to a multiple of 32 bytes, which is fine.  It will leave the `next-addr` to be 32-byte aligned.  But this is not sufficient to ensure that the second allocation gets its desired alignment!  The shmem allocator still needs add some padding _at the beginning of the second allocation_ to ensure that the second allocation request gets its desired alignment.

So, my point is that this code does nothing to actually achieve the alignment of the requested allocations.  There's no reason to round up allocation size requests.  Instead, the underlying shmem allocator itself needs to adjust the pointers it returns to ensure that they are aligned.